src/client.js

```diff
@@ -500,6 +500,16 @@ Client.prototype.more = function(data) {
 	};
 };
 
+Client.prototype.search = function(query) {
+	if (this.messageStorage) {
+		for (const storage of this.messageStorage) {
+			if (storage.database) {
```

Comment:
Storage is supposed to be opaque to TL, for example load message has a `canProvideMessages` helper.

https://github.com/thelounge/thelounge/blob/6f7444dfe38f71310bd22147f431e1273182e865/src/models/chan.js#L243